### PR TITLE
adjust redirects to respect possible subdirectory configuration

### PR DESCRIPTION
**Pull request by machisuji**

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -30,15 +30,17 @@
...
   # Redirect deprecated issue links to new work packages uris
-  match '/issues(/)'    => redirect('/work_packages/')
+  match '/issues(/)'    => redirect("#{dir}/work_packages/")
   # The URI.escape doesn't escape / unless you ask it to.
   # see https://github.com/rails/rails/issues/5688
-  match '/issues/*rest' => redirect { |params, req| "/work_packages/#{URI.escape(params[:rest])}" }
+  match '/issues/*rest' => redirect { |params, req| "#{dir}/work_packages/#{URI.escape(params[:rest])}" }
 
   # Redirect wp short url for work packages to full URL
   match '/wp(/)'    => redirect('/work_packages/')
```
Comment (hschink) [inline config/routes.rb]: @machisuji I think this is missing an `{dir}` too.

Comment (machisuji) [inline config/routes.rb]: Indeed it is.

**Comment by ulferts** [comment]
I don't like the code to much but granted, I don't have a better solution. 

Because I am gruntled, I am nitpicking:

This could be improved by wrapping the redirect [statements in scopes](http://guides.rubyonrails.org/routing.html) like this:

<pre>
relative_url_root = OpenProject::Configuration['rails_relative_url_root'] || ''
scope relative_url_root do
  match '/issues(/)'    => redirect("/work_packages/")
end
</pre>

**Comment by machisuji** [comment]
Wouldn't that be along the lines of the [alternative solution](https://github.com/opf/openproject/compare/feature;subdirectory_routing) to this whole problem which was to include the relative url as a scope? But that was met with disfavour as well IIRC. /cc @cratz

**Comment by ulferts** [comment]
What I dislike about the alternative solution is that it hides it's magic somewhere in a initialiser. And I am not even sure if the initialiser is loaded in time.

**Comment by machisuji** [comment]
```
relative_url_root = OpenProject::Configuration['rails_relative_url_root'] || ''

scope relative_url_root do
  match '/issues(/)'    => redirect("/work_packages/")
end
```

This does not work. The route cannot be found. Which makes sense seeing as the scope you want to use there is removed from the path by rack, remember?

I.e. `GET /open_project/issues/` leads to just `/issues` which then cannot be found since the route is expecting the previously dropped open_project prefix.

**Comment by ulferts** [comment]
you are right of course. But please at least rename the used variable from "dir" to something more speaking

**Comment by ulferts** [comment]
closed in favor of #973

**Pull request closed without merging**